ethereum-consensus/src/capella/block_processing.rs

```diff
@@ -37,10 +41,33 @@ pub fn process_bls_to_execution_change<
         BYTES_PER_LOGS_BLOOM,
         MAX_EXTRA_DATA_BYTES,
     >,
-    _signed_address_change: &mut SignedBlsToExecutionChange,
-    _context: &Context,
+    signed_address_change: &mut SignedBlsToExecutionChange,
+    context: &Context,
 ) -> Result<()> {
-    unimplemented!()
+    let address_change = signed_address_change;
+
+    assert!(address_change.message.validator_index < state.validators.len());
+
+    let validator = &mut state.validators[address_change.message.validator_index];
+
+    assert!(validator.withdrawal_credentials.starts_with(&[BLS_WITHDRAWAL_PREFIX]));
+    assert!(
+        validator.withdrawal_credentials[1..]
+            == hash(address_change.message.from_bls_public_key.as_ref())[1..]
+    );
+
+    let domain =
+        compute_domain(BlsToExecutionChange, None, Some(state.genesis_validators_root), context)?;
+    let signing_root = compute_signing_root(address_change, domain)?;
```

Comment:
this needs `&mut` to be able to compute the `hash_tree_root`; also, the message being signed over is the `message` here, not the full data

```suggestion
    let signing_root = compute_signing_root(&mut address_change.message, domain)?;
```

Reply:
If we follow the spec and do

```rust
    let address_change = &signed_address_change.message;
```

we need to use the `signed_address_change` here and run into ownership issues down the line , and this consume the var.

We also can't clone it :(

Reply:
where is it an issue?

we should be able to separate out the message from the signature without any problems

```rust
let address_change = &mut signed_address_change.message;
let signature = &signed_address_change.signature;
```

should be fine as these are disjoint field accesses

Reply:
done